src/sagemaker/jumpstart/curated_hub/curated_hub.py

```diff
@@ -113,3 +173,241 @@ def delete_model(self, model_name: str, model_version: str = "*") -> None:
+                "and optional 'version'.",
+            )
+
+        # Retrieve latest version of unspecified JumpStart model versions
+        model_version_list = []
+        for model in model_list:
+            version = model.get("version", "*")
+            if version == "*":
+                model = self._populate_latest_model_version(model)
+                JUMPSTART_LOGGER.warning(
+                    "No version specified for model %s. Using version %s",
+                    model["model_id"],
+                    model["version"],
+                )
+            model_version_list.append(JumpStartModelInfo(model["model_id"], model["version"]))
+
+        js_models_in_hub = self._get_jumpstart_models_in_hub()
+
+        models_to_sync = self._determine_models_to_sync(model_version_list, js_models_in_hub)
+        JUMPSTART_LOGGER.warning(
+            "Syncing the following models into Hub %s: %s", self.hub_name, models_to_sync
+        )
+
+        # Delete old models?
+
+        # CopyContentWorkflow + `SageMaker:ImportHubContent` for each model-to-sync in parallel
+        tasks: List[futures.Future] = []
+        with futures.ThreadPoolExecutor(
+            max_workers=self._default_thread_pool_size,
+            thread_name_prefix="import-models-to-curated-hub",
+        ) as deploy_executor:
+            for thread_num, model in enumerate(models_to_sync):
+                task = deploy_executor.submit(self._sync_public_model_to_hub, model, thread_num)
+                tasks.append(task)
+
+        # Handle failed imports
+        results = futures.wait(tasks)
+        failed_imports: List[Dict[str, Any]] = []
+        for result in results.done:
+            exception = result.exception()
+            if exception:
+                failed_imports.append(
+                    {
+                        "Exception": exception,
+                        "Traceback": "".join(
+                            traceback.TracebackException.from_exception(exception).format()
+                        ),
+                    }
+                )
+        if failed_imports:
+            raise RuntimeError(
+                f"Failures when importing models to curated hub in parallel: {failed_imports}"
+            )
+
+    def _sync_public_model_to_hub(self, model: JumpStartModelInfo, thread_num: int):
+        """Syncs a public JumpStart model version to the Hub. Runs in parallel."""
+        model_specs = utils.verify_model_region_and_return_specs(
+            model_id=model.model_id,
+            version=model.version,
+            region=self.region,
+            scope=JumpStartScriptScope.INFERENCE,
+            sagemaker_session=self._sagemaker_session,
+        )
+        studio_specs = self._fetch_studio_specs(model_specs=model_specs)
+
+        dest_location = S3ObjectLocation(
+            bucket=self.hub_bucket_name, key=f"{model.model_id}/{model.version}"
```

Comment:
didn't we say we would put "curated models" under a dedicated prefix in the Hub Bucket?

Something like 
`s3://<hub_bucket_name>/<hub_prefix_with_timestamp>/curated-models/<model-id>/<model-version>`

Reply:
I don't think so, and I don't see the immediate benefit tbh

Reply:
Immediate benefit maybe not, but there is a theoretical case where the customer uses the same Hub for their own private model use cases and curated models. In this case, having the delineation of `curated-models` makes it far easier to root cause issues